code/go/0chain.net/smartcontract/storagesc/models_test.go

```diff
@@ -121,3 +121,86 @@ func TestStorageAllocation_filterBlobbers(t *testing.T) {
 	list[1].Capacity, list[1].Used = 330, 100
 	assert.Len(t, alloc.filterBlobbers(list, now, size), 2)
 }
+
+func TestStorageAllocation_diversifyBlobbers(t *testing.T) {
+	type args struct {
+		params []int
+		size int
+	}
+
+	type want struct {
+		params []int
+	}
+
+	var params = []*StorageNode{
+		&StorageNode{Geolocation:
+			StorageNodeGeolocation{Latitude: 37.773972, Longitude: -122.431297}}, // San Francisco
+		&StorageNode{Geolocation:
+			StorageNodeGeolocation{Latitude: -33.918861, Longitude: 18.423300}}, // Cape Town
+		&StorageNode{Geolocation:
+			StorageNodeGeolocation{Latitude: 59.937500, Longitude: 30.308611}}, // St Petersburg
+		&StorageNode{Geolocation:
+			StorageNodeGeolocation{Latitude: 22.633333, Longitude: 120.266670}}, // Kaohsiung City
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
```

Comment:
What happens if the StorageNodes have no Latitude or Longitude information. Maybe that information is optional for blobbers. Anyway I think we should add a unit test for that.

Reply:
If not set, there will be zeros.
I don't think "randomizing" make sense. It's hard to define what and when to randomize. For example, iff 3 of 5 geolcations are "zeros"? What should we do?

Frankly, i wouldn't overcomplicated it with randomizing. And it's not about geo (!) diversification, anyway.

Reply:
So what do we do? At the moment the code tries to diversify a set of blobbers with zero longitudes and latitude, which seems a somewhat inefficient and uncontrolled way of randomising them.

Reply:
Imho, a test with all the zeros makes no sense, as the result is basically defined by order of the blobs in the array in such case. The test wouldn't prove/disprove the calcs.

Reply:
If someone just uses this change without their blobbers then all zeros is what we will be getting. An all-zero test makes sure the method does not error out but does something sensible.